.github/workflows/build-deploy-ec2.yml

```diff
@@ -0,0 +1,65 @@
+name: Build and Deploy to EC2
+on: workflow_dispatch
+jobs:
+  build-deploy:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10"]
+        node-version: [16.x]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install Python dependencies
+        working-directory: ./api
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install flake8 tox
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+          if [ -f test-requirements.txt ]; then pip install -r test-requirements.txt; fi
+      - name: Lint with flake8
+        working-directory: ./api
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+      - name: Test with tox
+        working-directory: ./api
+        run: |
+          tox
```

Comment:
Is this the successful build log? `https://github.com/hackforla/HomeUniteUs/actions/runs/5651696696/job/15310200967`

If so, I can't seem to find the `tox` step. Am I overlooking it? I'm mostly wondering because the `api` test cases were failing on my machine. I'd expect them to fail here to, unless I'm missing something

Reply:
🤦‍♂️That link I posted was the PR build link haha. 

Have you tried out the workflow and confirmed that it worked? If so, can I take a look at the deployment log?

Thanks!

Reply:
:blush:  Actually, I'm currently figuring out how to test it without merging to the main branch.

Reply:
Have you tried out this workflow yet? It might be worth sending off a build before approving, to ensure the SSH access is working. 

Do we have a development EC2 instance? If we did, then we wouldn't have to worry about breaking the current site

Reply:
@Joshua-Douglas Yes, I've got it in a testable state and have been modifying it. You can see all of the runs in the Actions tab.

At this time, we have just one EC2 instance. It hosts the static landing page for https://homeunite.us/ and acts as the dev/test/demo host for https://dev.homeunite.us. As of this writing, it's hosting an old version of the frontend and the backend is down.